pkg/controller/cronjob/cronjob_controller.go

```diff
@@ -148,6 +149,23 @@ func SyncOne(sj batch.CronJob, js []batch.Job, now time.Time, jc jobControlInter
 			recorder.Eventf(&sj, v1.EventTypeNormal, "SawCompletedJob", "Saw completed job: %v", j.Name)
 		}
 	}
+
+	// Remove any job reference from the active list if the corresponding job does not exist any more.
+	// Otherwise, the cronjob may be stuck in active mode forever even though there is no matching
+	// job running.
+
+	childrenJobs := make(map[types.UID]bool)
+	for _, childJob := range js {
+		childrenJobs[childJob.ObjectMeta.UID] = true
+	}
+	for _, j := range sj.Status.Active {
+		found := childrenJobs[j.UID]
+		if !found {
+			recorder.Eventf(&sj, v1.EventTypeNormal, "MissingJob", "Active job went missing: %v", j.Name)
+			deleteFromActiveList(&sj, j.UID)
+		}
+	}
+
```

Comment:
I am a bit confused as to what you mean -- when a job is deleted manually as in this issue, it will not be in `js` at all, so we will never actually hit it in the above loop? 

This is why I instead iterate over the active list directly, and look for jobs that are missing in `js`.